src/util/tubemap.test.js

```diff
@@ -211,4 +214,394 @@ describe('cigar_string', () => {
       };
       expect(cigar_string(readPath)).toBe("0M");
     });
-  });
\ No newline at end of file
+  });
+
+
+// Test to make sure that a node and a set of reads make sense together.
+function checkNodeExample(node, reads) {
+  let nodeName = null
+  // Set or check node name for the given visit of the given read
+  function checkNodeName(read, visitIndex) {
+    let visit = read.sequenceNew[visitIndex]
+    if (nodeName) {
+      if (nodeName !== visit.nodeName) {
+        throw new Error("Different visits to this node have different node names! " + nodeName + " vs. " + visit.nodeName)
+      }
+    } else {
+      // See if we can pick up a node name
+      nodeName = visit.nodeName
+    }
+  }
+  for (let [readNum, visitIndex] of node.incomingReads) {
+    if (readNum >= reads.length) {
+      throw new Error("Incoming read " + readNum + " doesn't exist")
+    }
+    let read = reads[readNum]
+    if (visitIndex >= read.sequenceNew.length) {
+      throw new Error("Incoming read " + readNum + " visit " + visitIndex + " doesn't exist")
+    }
+    if (visitIndex == 0) {
+      // This shouldn't happen a lot because it means the read started before the first node in the region.
+      if (read.firstNodeOffset !== undefined && read.firstNodeOffset !== 0) {
+        // This shouldn't happen ever; we can't enter this first node anywhere but the start.
+        throw new Error("Read is entering a node first but has a nonzero first node offset!")
+      }
+    }
+    checkNodeName(read, visitIndex)
+    if (visitIndex == read.sequenceNew.length - 1) {
+      // Read ends here
+      if (read.finalNodeCoverLength > node.sequenceLength) {
+        throw new Error("Final node cover length too long")
+      }
+    }
+  }
+  for (let readNum of node.internalReads) {
+    if (readNum >= reads.length) {
+      throw new Error("Internal read " + readNum + " doesn't exist")
+    }
+    let read = reads[readNum]  
+    // Internal reads can only have one visit
+    let visitIndex = 0
+    if (read.sequenceNew.length !== 1) {
+      throw new Error("Internal reads can only visit one node, but read" + readNum + " doesn't")
+    } 
+    checkNodeName(read, visitIndex)
+  }
+
+  for (let [readNum, visitIndex] of node.outgoingReads) {
+    if (readNum >= reads.length) {
+      throw new Error("Outgoing read " + readNum + " doesn't exist")
+    }
+    let read = reads[readNum]
+    if (visitIndex >= read.sequenceNew.length) {
+      throw new Error("Outgoing read " + readNum + " visit " + visitIndex + " doesn't exist")
+    }
```

Comment:
We should constrain the outgoing reads to start at that node, since that's the semantics actually used.
```suggestion
    }
    if (visitIndex !== 0) {
      throw new Error("Outgoing read did not start at the node it is outgoing for")
    }
```